game/songparser.cc

```diff
@@ -308,4 +275,31 @@ void SongParser::finalize () {
 		// Add song beat markers
 		for (unsigned ts = 0; ts < m_tsEnd; ts += m_tsPerBeat) { m_song.beats.push_back (tsTime (ts)); }
 	}
-}
\ No newline at end of file
+}
+
+SongParser::BPM SongParser::getBPM(double ts) const {
+	for (auto& itb: boost::adaptors::reverse(m_bpms)) {
+		if (itb.begin <= ts) return itb;
```

Comment:
In the original code of `getBPM` one had the line `if (itb.begin >= ts && ts != 0) continue;`
This crashed indeed telling the user no bpm was found.

@Lord-Kamina looked at this and noticed that changing the if to `if (itb.begin > ts && ts != 0) continue;` solved the problem.

Reply:
Also as in current master this method looks as this
```
BPM getBPM (double ts) {
	for (auto& itb : boost::adaptors::reverse (m_bpms)) {
			if (itb.begin > ts) { 
				continue; 
			} else {
				return itb;
			}
		}
		throw std::runtime_error("No BPM definition prior to this note...");
	}
}
```

In yours (@Tronic ) i don't see any continue in the loop hmmm?

Reply:
@Baklap4 The new code has that if clause reversed so that return comes first. After returning, no else block is needed, and continue does nothing because it would be the last statement of the loop.

The code in your message is misindented and has an extra } (master appears to be OK - throw is outside of the loop).